### ISSUE TITLE
Potential bug in triplet loss implementation

### ISSUE THREAD

**Issue by vilhess** [jinnnju/PaAno#1]
Hi, thank you for the paper and for releasing the code!

I think there may be a bug in the triplet loss implementation. When the diagonal is masked with `-inf` in similarity space and then converted to distances via `1 - sim`, those entries become `+inf`:
```
1 - (-inf) = +inf
```

Since `torch.max` is then applied in distance space, it will always select these `+inf` diagonal entries as the "hardest" negatives — meaning the loss becomes `relu(pos_dist - inf + 0.5) = 0` for every sample.
```python
_sim_ap_f = _sim_ap.clone()
_sim_ap_f.diagonal().fill_(-float('inf'))
neg_dists = 1 - _sim_ap_f                        # diagonal becomes +inf
hard_neg_dists, _ = torch.max(neg_dists, dim=1)  # always picks diagonal
pos_dists = 1 - pos_sims
triplet_loss = F.relu(pos_dists - hard_neg_dists + 0.5).mean()  # always 0?
```

Am I misreading the code, or is this indeed a bug? (I think the solution is to fill instead with  `+inf`)

**Comment by jinnnju** [comment]
Hi, thank you for your interest in our paper and for taking the time to examine the code!

The issue you pointed out was identified as a bug and has now been corrected in the repository. 

```
_sim_ap_f.diagonal().fill_(+float('inf')) # fixed bug
```

Thank you for helping us catch this!
